### PR TITLE
R11BRT-777 .NET 8 Hosting Bundle

### PR DESCRIPTION
**Pull request by ajps28**
- Update Hosting Bundle to .NET 8 (8.0.0)
- Adapt unit tests and add new test cases for .NET 8 version
- Refactor unit tests for readability
- Raise version

### CODE REVIEW

**Review by pedro132 — CHANGES_REQUESTED** [review]
Missing a line in `Get-OSServerPreReqs.ps1` when the function is invoked passing the version using only with major (11). Otherwise, minor nitpicks.

File: src/Outsystems.SetupTools/Functions/Get-OSServerPreReqs.ps1 [hunk 1]
```diff
@@ -173,6 +173,15 @@ function Get-OSServerPreReqs
                     $requireDotNetCoreHostingBundle3 = $true
                     $requireDotNetHostingBundle6 = $true
```
[the remark this answers is not in the document]
Comment (ajps28) [reply src/Outsystems.SetupTools/Functions/Get-OSServerPreReqs.ps1:174]: Good catch, corrected now.

File: test/unit/Install-OSServerPreReqs.Tests.ps1 [hunk 2]
```diff
@@ -1337,6 +1630,33 @@ InModuleScope -ModuleName OutSystems.SetupTools {
...
+
+            $result = Install-OSServerPreReqs -MajorVersion '11' -MinorVersion '25' -PatchVersion '1' -RemovePreviousHostingBundlePackages $true -ErrorVariable err -ErrorAction SilentlyContinue
+
+            It 'Should run the .NET installation' { Assert-MockCalled @assRunInstallDotNet }
+            It 'Should run the BuildTools installation' { Assert-MockCalled @assRunInstallBuildTools }
+            It 'Should install the windows features installation' { Assert-MockCalled @assRunInstallWindowsFeatures }
+            It 'Should not run the .NET core 2.1 installation' { Assert-MockCalled @assNotRunInstallDotNetCore21 }
+            It 'Should not run the .NET core installation' { Assert-MockCalled @assNotRunInstallDotNetCore }
+            It 'Should run the .NET 6.0 Hosting Bundle installation' { Assert-MockCalled @assNotRunInstallDotNetHostingBundle }
+            It 'Should not run the .NET 8.0 Hosting Bundle installation' { Assert-MockCalled @assRunInstallDotNetHostingBundle8 }
```
[the remark this answers is not in the document]
Comment (ajps28) [reply test/unit/Install-OSServerPreReqs.Tests.ps1]: nice catches, thanks!

File: test/unit/Install-OSServerPreReqs.Tests.ps1 [hunk 3]
```diff
@@ -1259,6 +1522,33 @@ InModuleScope -ModuleName OutSystems.SetupTools {
...
+
+            $result = Install-OSServerPreReqs -MajorVersion '11' -MinorVersion '25' -PatchVersion '1' -ErrorVariable err -ErrorAction SilentlyContinue
+
+            It 'Should run the .NET installation' { Assert-MockCalled @assRunInstallDotNet }
+            It 'Should run the BuildTools installation' { Assert-MockCalled @assRunInstallBuildTools }
+            It 'Should install the windows features installation' { Assert-MockCalled @assRunInstallWindowsFeatures }
+            It 'Should not run the .NET core 2.1 installation' { Assert-MockCalled @assNotRunInstallDotNetCore21 }
+            It 'Should not run the .NET core installation' { Assert-MockCalled @assNotRunInstallDotNetCore }
+            It 'Should run the .NET 6.0 Hosting Bundle installation' { Assert-MockCalled @assNotRunInstallDotNetHostingBundle }
+            It 'Should not run the .NET 8.0 Hosting Bundle installation' { Assert-MockCalled @assRunInstallDotNetHostingBundle8 }
```
[the remark this answers is not in the document]
Comment (ajps28) [reply test/unit/Install-OSServerPreReqs.Tests.ps1]: nice catches, thanks!

**Review by pedro132 — APPROVED** [review]
Ship it :shipit:

**Pull request merged**